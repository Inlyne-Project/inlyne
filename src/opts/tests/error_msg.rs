macro_rules! snapshot_config_parse_error {
    ( $( ($test_name:ident, $md_text:ident) ),* $(,)? ) => {
        $(
            #[test]
            fn $test_name() {
                let err = ::toml::from_str::<$crate::opts::Config>($md_text).unwrap_err();

                ::insta::with_settings!({
                    description => $md_text,
                }, {
                    ::insta::assert_display_snapshot!(err);
                });
            }
        )*
    }
}

const UNKNOWN_THEME: &str = r#"light-theme.code-highlighter = "doesnt-exist""#;
const INVALID_THEME_TY: &str = "light-theme.code-highlighter = []";

snapshot_config_parse_error!(
    (unknown_theme, UNKNOWN_THEME),
    (invalid_theme_ty, INVALID_THEME_TY),
);
