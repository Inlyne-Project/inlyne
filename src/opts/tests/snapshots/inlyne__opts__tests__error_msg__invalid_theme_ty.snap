---
source: src/opts/tests/error_msg.rs
description: "light-theme.code-highlighter = []"
expression: err
---
TOML parse error at line 1, column 32
  |
1 | light-theme.code-highlighter = []
  |                                ^^
Expects either a default theme name or a path to a custom theme. E.g.
default: "inspired-github"
custom:  { path = "/path/to/custom.tmTheme" }

